DigitalLearningSolutions.Data.Tests/Services/RegistrationServiceTests.cs

```diff
@@ -1102,6 +1103,97 @@ public void CreateDelegateAccountForExistingUser_with_unapproved_IP_registers_de
+                            A<DelegateRegistrationModel>._,
+                            A<int>._,
+                            A<DateTime>._,
+                            A<IDbTransaction>._
+                        )
+                )
+                .MustNotHaveHappened();
+            A.CallTo(
+                () => registrationDataService.ReregisterDelegateAccountAndCentreDetailForExistingUser(
+                    A<DelegateRegistrationModel>._,
+                    A<int>._,
+                    A<int>._,
+                    A<DateTime>._
+                )
+            ).MustNotHaveHappened();
+            var expectedError =
+                "Could not create account for delegate on registration. " +
+                $"Failure: active delegate account with ID {existingDelegateId} already exists " +
+                $"at centre with ID {model.Centre} for user with ID {userId}";
+            act.Should().Throw<DelegateCreationFailedException>().WithMessage(expectedError);
+        }
+
+        [Test]
+        public void
+            CreateDelegateAccountForExistingUser_reregisters_delegate_if_user_already_has_inactive_delegate_at_centre()
+        {
+            // Given
+            const int userId = 2;
+            const int existingDelegateId = 5;
+            var model = RegistrationModelTestHelper.GetDefaultInternalDelegateRegistrationModel();
+            var currentTime = DateTime.Now;
+            A.CallTo(() => clockService.UtcNow).Returns(currentTime);
+            GivenUserEntityExistsWithDelegate(userId, existingDelegateId, model.Centre, false);
+
+            // When
+            registrationService.CreateDelegateAccountForExistingUser(
+                model,
+                userId,
+                "::1",
+                false
+            );
+
+            // Then
+            A.CallTo(
+                    () =>
+                        registrationDataService.RegisterDelegateAccountAndCentreDetailForExistingUser(
+                            A<DelegateRegistrationModel>._,
+                            A<int>._,
+                            A<DateTime>._,
+                            A<IDbTransaction>._
+                        )
+                )
+                .MustNotHaveHappened();
+            A.CallTo(
+                () => registrationDataService.ReregisterDelegateAccountAndCentreDetailForExistingUser(
+                    A<DelegateRegistrationModel>.That.Matches(
+                        d => d.Centre == model.Centre && d.Approved && d.Active && d.Answer1 == model.Answer1 &&
+                             d.Answer2 == model.Answer2 && d.Answer3 == model.Answer3 && d.Answer4 == model.Answer4 &&
+                             d.Answer5 == model.Answer5 && d.Answer6 == model.Answer6
+                    ),
+                    userId,
+                    existingDelegateId,
+                    currentTime
+                )
+            ).MustHaveHappenedOnceExactly();
+        }
+

```

Comment:
At least one of the other successful `CreateDelegateAccountForExistingUser` tests should check that a call to `registrationDataService.ReregisterDelegateAccountAndCentreDetailForExistingUser` `MustNotHaveHappened`

Reply:
I've added them to all the successful new account registrations, because why not.